tests/common/test_cgroupconfigurator.py

```diff
@@ -31,29 +31,105 @@
 class CGroupConfiguratorSystemdTestCase(AgentTestCase):
     @classmethod
     def tearDownClass(cls):
-        CGroupConfigurator._instance = None # pylint: disable=protected-access
+        # protected-access<W0212> Disabled: OK to access CGroupConfigurator._instance from unit test for CGroupConfigurator
+        CGroupConfigurator._instance = None  # pylint: disable=protected-access
         AgentTestCase.tearDownClass()
 
     @staticmethod
-    def _get_new_cgroup_configurator_instance(initialize=True):
-        CGroupConfigurator._instance = None # pylint: disable=protected-access
+    def _get_new_cgroup_configurator_instance(initialize=True, mock_commands=None, mock_files=None):
+        # protected-access<W0212> Disabled: OK to access CGroupConfigurator._instance from unit test for CGroupConfigurator
+        CGroupConfigurator._instance = None  # pylint: disable=protected-access
         configurator = CGroupConfigurator.get_instance()
+        CGroupsTelemetry.reset()
         if initialize:
-            with mock_cgroup_commands():
+            with mock_cgroup_commands() as mocks:
+                if mock_files is not None:
+                    for item in mock_files:
+                        mocks.add_file(item[0], item[1])
+                if mock_commands is not None:
+                    for command in mock_commands:
+                        mocks.add_command(command[0], command[1])
                 configurator.initialize()
         return configurator
 
     def test_initialize_should_start_tracking_the_agent_cgroups(self):
-        CGroupConfiguratorSystemdTestCase._get_new_cgroup_configurator_instance()
-
-        tracked = CGroupsTelemetry._tracked # pylint: disable=protected-access
+        configurator = CGroupConfiguratorSystemdTestCase._get_new_cgroup_configurator_instance()
 
-        self.assertTrue(
-            any(cg for cg in tracked if cg.name == 'walinuxagent.service' and 'cpu' in cg.path),
-            "The Agent's CPU is not being tracked")
-        self.assertTrue(
-            any(cg for cg in tracked if cg.name == 'walinuxagent.service' and 'memory' in cg.path),
-            "The Agent's memory is not being tracked")
+        # protected-access<W0212> Disabled: OK to access CGroupConfigurator._tracked from unit test for CGroupConfigurator
+        tracked = CGroupsTelemetry._tracked  # pylint: disable=protected-access
```

Comment:
Is this protected access ok here? It would be nice to keep unit tests agnostic of the internals of our prod modules.

Reply:
Yes, in an ideal situation. I believe in this situation it is OK to break the rule; the unit tests do not make heavy use of internals, and this is unit test for A using A's internals. Things get dicey in other parts of the code where we have unit tests for A using internals of D when A uses B uses C uses D.
Exposing the internals of CGroupsTelemetry just the test's benefit doesn't seem right, and looking more more complicated way to do this check doesn't seem to be worth it.